wheat_filtration/mallet.py

```diff
@@ -1,68 +1,80 @@
 from collections import OrderedDict
 import os
-import shlex
 import subprocess
+import warnings
 
 import gensim.corpora as corpora
 from gensim.models.wrappers import LdaMallet
 import numpy as np
 from nltk.corpus import stopwords
+from scipy.sparse import coo_matrix
 
 import munge
 
+# not sure what to do with this... vs code keeps moving it
+# do i even need to keep it in? can we just assume theyve done this?
+# import nltk
+# nltk.download('stopwords')
 
-MALLET_PATH = None  # "/Users/fauma/Mallet-master/bin/mallet"  #
 
+MALLET_PATH = "/Users/fauma/Mallet-master/bin/mallet"  # None  #
 
-def _call_command_line(string):
-    """Executes string as a command line prompt."""
-    return subprocess.call(shlex.split(string))
+
+def _call_command_line(string, **kwargs):
+    """Executes string as a command line prompt. stdout and stderr are keyword args."""
+    return subprocess.run(string, **kwargs)
 
 
 def _make_doc_dictionary(path_to_mallet, mallet_instance_filepath):
     """Returns an Ordered Dictionary containing document unique IDs as keys and
     preprocessed document text as values, for all documents in the corpus.
     Creates and deletes a temporary file in the current directory called "temp_docs_will_be_deleted.txt". """
-    _call_command_line("{} info --input {} --print-instances > \
-        temp_docs_will_be_deleted.txt".format(path_to_mallet,
-                                              mallet_instance_filepath))  # couldnt find a way to do this without piping
+    command = "{} info --input {} --print-instances".format(path_to_mallet,
+                                                            mallet_instance_filepath)
+    outfile = open("temp_docs_will_be_deleted.txt", "w")
+    _call_command_line(command.split(" "), stdout=outfile)
+    outfile.close()
+
     docs_dictionary = OrderedDict()
     with open("temp_docs_will_be_deleted.txt", "r") as in_file:
         prev_line = ""
-        curr_doc_id = ""
-        curr_doc = []
+        current_doc_id = ""
+        current_doc = []
         for line in in_file:
             if line.strip() == "":
-                docs_dictionary[curr_doc_id] = " ".join(curr_doc)
-                curr_doc = []
+                docs_dictionary[current_doc_id] = " ".join(current_doc)
+                current_doc = []
             elif prev_line.strip() == "":
-                curr_doc_id = line.split()[0]
+                current_doc_id = line.split()[0]
             else:
-                curr_doc.append(line.split()[1])
+                current_doc.append(line.split()[1])
             prev_line = line
     os.remove("temp_docs_will_be_deleted.txt")
     return docs_dictionary
 
 
 def _make_wordcount_and_vocab(mallet_topic_wordcount_filepath, n_topics):
-    """Returns a tuple containing a matrix of topic wordcounts, the corpus vocabulary,
+    """Returns a tuple containing a CSC sparse matrix of topic wordcounts, the corpus vocabulary,
```

Comment:
i used CSC because i thought it made sense for the calculations we would use the matrix for in the filtration step (summing along axis 0) but i don't actually know if this makes sense haha. COO is more versatile right? so it probably makes sense to store it in that format so users can use it however they like..